mountpoint-s3/src/prefetch.rs

```diff
@@ -325,16 +326,17 @@ where
             return Ok(());
         }
 
+        let mut rowgroup_col_cache = self.data_cache.write().await;
+        *rowgroup_col_cache = Some(HashMap::new());
+        drop(rowgroup_col_cache);
```

Comment:
nit: consider using scoping instead of explicit `drop`. E.g.:
```suggestion
        {
            let mut rowgroup_col_cache = self.data_cache.write().await;
            *rowgroup_col_cache = Some(HashMap::new());
        }
```
Or in this case even just:
```suggestion
        *(self.data_cache.write().await) = Some(HashMap::new());
```